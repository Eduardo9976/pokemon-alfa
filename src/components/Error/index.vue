<template>
  <div class="error">
    <div>
      <img src="../../assets/images/warning.svg" alt="Atenção" />
      <p>
        Ops...Serviço Indisponível no momento. Por favor, tente novamente mais
        tarde.
      </p>
    </div>
  </div>
</template>

<style>
.error {
  min-height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}
.error div {
  background: rgba(0, 0, 0, 0.5);
  padding: 40px;
  border-radius: 4px;
  border: 3px solid rgba(0, 0, 0, 0.7);
}
.error p {
  font-family: 20px;
  color: #ffffff;
}
.error i {
  font-size: 30px;
}
.error img {
  max-width: 100px;
  margin: 0 auto;
  height: auto;
  display: block;
  margin-bottom: 20px;
}
</style>
