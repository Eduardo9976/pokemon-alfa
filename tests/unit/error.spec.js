import Error from '../../src/components/Error'
import { shallowMount } from '@vue/test-utils'

describe('<Error />', () => {
    it('should render correctly component', () => {
        const wrapper = shallowMount(Error, {
            
        })
        expect(wrapper.html()).toMatchSnapshot()
        expect(wrapper.find('.error').exists()).toBe(true)
        expect(wrapper.find('img').exists()).toBe(true)
        expect(wrapper.find('p').exists()).toBe(true)
    })
    it('should render an error message', () => {
        const wrapper = shallowMount(Error, {
            
        })
        expect(wrapper.find('p').text()).toBe('Ops...Serviço Indisponível no momento. Por favor, tente novamente mais tarde.')
        expect(wrapper.find('img').attributes('alt')).toBe('Atenção')
    })  
})