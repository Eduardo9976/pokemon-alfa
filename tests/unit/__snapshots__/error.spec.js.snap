// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render correctly component 1`] = `
<div class="error">
  <div><img src="" alt="Atenção">
    <p> Ops...Serviço Indisponível no momento. Por favor, tente novamente mais tarde. </p>
  </div>
</div>
`;
